### PR TITLE
[GEOT-6808] -  Http headers set for WMTS isn't used for GetCapabiliti…

### PR DESCRIPTION
**Pull request by roarbra**
The Headers for HTTP must be set during the constructor, because there is a call for GetCapabilities therein.

It will break build downstream, because Geoserver are using a constructor with null for it's third argument. Now we are introducing two constuctors with similar signatures.

# Checklist

Reviewing is a process done by project maintainers, **mostly on a volunteer basis** (thus limited in time). We need to keep the review overhead as small as possible, and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [X] Confirm you have read the [contribution guidelines](https://github.com/geotools/geotools/blob/master/CONTRIBUTING.md)
- [X] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [X] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.
- [X] The changes are not causing two modules to share the same Java packages (to avoid [Java 9+ split package](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed) issues)
- [ ] The changes are not breaking the build in downstream projects using SNAPSHOT dependencies, GeoWebCache and GeoServer (there is an automatic PR check verifying this, check this when it turns green).

The following are required only for core and extension modules (they are welcomed, but not required, for unsupported modules):

- [X] There is an issue in [Jira](https://osgeo-org.atlassian.net/projects/GEOT) describing the bug/task/new feature (a notable exemptions is, changes not visible to end users). The ticket is for the GeoTools project, if the issue was found elsewhere it's a good practice to link to the origin ticket/issue.
- [X] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [X] PR for bug fixes and small new features are presented as a single commit
- [X] Commit message(s) must be in the form "[GEOT-XYZW] Title of the Jira ticket"
- [ ] New unit tests have been added covering the changes
- [ ] This PR passes all existing unit tests (test results will be reported by Continuous Integration after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html) (QA checks results will be reported by Continuous Integration after opening this PR)
- [ ] Documentation has been updated accordingly.

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or not applicable.

### CODE REVIEW

**Comment by aaime** [comment]
Ok so you are introducing a backwards incompatible change. I don't see an obvious way to avoid it, but since it's a breaking change, it must be reported in the ``upgrade.rst``, just like you did for the HTTPClient.

File: modules/extension/wmts/src/main/java/org/geotools/ows/wmts/WebMapTileServer.java [hunk 1]
```diff
@@ -186,6 +250,11 @@ public WMTSServiceType getType() {
         return type;
     }
 
+    /**
```
Comment (aaime) [inline modules/extension/wmts/src/main/java/org/geotools/ows/wmts/WebMapTileServer.java:253]: This should be reported in the upgrade guide too.

**Comment by aaime** [comment]
The rest of the changes looks good to me.

**Comment by roarbra** [comment]
Thanks @aaime ! I've updated the documentation.

File: docs/user/welcome/upgrade.rst [hunk 2]
```diff
@@ -211,6 +211,27 @@ In addition a new plugin ``gt-http-commons`` has been added for MultithreadedHtt
    HTTPClient http = HTTPClientFinder.createClient(hints);
    HTTPResponse response = http.get();
 
+WMTS - WebMapTileServer initialisation
+--------------------------------------
+
+We have introdused a new contructor for the WebMapTileServer.
```
Comment (mprins) [inline docs/user/welcome/upgrade.rst:217]:
```suggestion
We have introduced a new contructor for the WebMapTileServer.
```

File: docs/user/welcome/upgrade.rst [hunk 3]
```diff
@@ -211,6 +211,27 @@ In addition a new plugin ``gt-http-commons`` has been added for MultithreadedHtt
...
+See list of available constructors:
+
+.. code-block:: java
+
+  public WebMapTileServer(URL serverURL, HTTPClient httpClient)
+  public WebMapTileServer(URL serverURL, HTTPClient httpClient, Map<String, String> headers) // <- NEW CONSTRUCTOR
+  public WebMapTileServer(URL serverURL, HTTPClient httpClient, WMTSCapabilities capabilities)
+  public WebMapTileServer(URL serverURL, HTTPClient httpClient, WMTSCapabilities capabilities, Map<String, Object> hints)
+
+By the same reason we will not allow changes to the headers after initialisation,
```
Comment (mprins) [inline docs/user/welcome/upgrade.rst:231]:
```suggestion
For the same reason we will not allow changes to the headers after initialisation,
```

**Comment by aaime** [comment]
PR was not compiling due to a collection type mismatch. Stepped in to fix, as @jodygarnett is about to cut 25-RC (any API change past that would go in 26.x by default, to be released in September, unless one can convince the PSC otherwise and backport).

```
diff --git a/modules/extension/wmts/src/test/java/org/geotools/ows/wmts/map/WMTSCoverageReaderTest.java b/modules/extension/wmts/src/test/java/org/geotools/ows/wmts/map/WMTSCoverageReaderTest.java
index 767db6a982..346fa5cddd 100644
--- a/modules/extension/wmts/src/test/java/org/geotools/ows/wmts/map/WMTSCoverageReaderTest.java
+++ b/modules/extension/wmts/src/test/java/org/geotools/ows/wmts/map/WMTSCoverageReaderTest.java
@@ -241,7 +241,7 @@ public class WMTSCoverageReaderTest {
                 templateURL);
     }
 
-    protected List<Tile> testInitMapRequest(WMTSCoverageReader wcr, ReferencedEnvelope bbox)
+    protected Set<Tile> testInitMapRequest(WMTSCoverageReader wcr, ReferencedEnvelope bbox)
             throws Exception {
 
         int width = 400;
```

**Pull request merged**